book/src/pong-tutorial/pong-tutorial-01.md

```diff
@@ -121,95 +135,153 @@ say "Pong!" instead of the sad, lowercase default of "pong".
+
+        graph_builder
+    }
+}
 ```
 
 The important thing to know right now is that this renders a black background.
 If you want a different color you can tweak the RGBA values inside the
-`.clear_target` method. Values range from 0.0 to 1.0, so to get that cool green
-color we started with back then, for instance, you can try
+`ExampleGraph`'s `builder` method. Values range from 0.0 to 1.0, so to get that cool green color we started with back then, for instance, you can try
 `[0.00196, 0.23726, 0.21765, 1.0]`.
 
-Now let's pack everything up and run it:
+Now let's pack everything up and run it back in the `main()` function:
 
 ```rust,edition2018,no_run,noplaypen
-# extern crate amethyst;
-# use amethyst::prelude::*;
-# use amethyst::renderer::{DisplayConfig, DrawFlat2D, Pipeline,
-#                        RenderBundle, Stage};
-# fn main() -> amethyst::Result<()> {
-# let path = "./resources/display_config.ron";
-# let config = DisplayConfig::load(&path);
-# struct Pong;
-# impl SimpleState for Pong { }
-let pipe = Pipeline::build()
-    .with_stage(
-#        Stage::with_backbuffer()
-#          .clear_target([0.0, 0.0, 0.0, 1.0], 1.0)
-#          .with_pass(DrawFlat2D::new()),
-    // --snip--
-    );
-
-let game_data = GameDataBuilder::default()
-    .with_bundle(
-      RenderBundle::new(pipe, Some(config))
-        .with_sprite_sheet_processor()
-    )?;
-
-let mut game = Application::new("./", Pong, game_data)?;
-
-game.run();
-# Ok(())
-# }
+    let game_data = GameDataBuilder::default()
+        // The WindowBundle provides all the scaffolding for opening a window and drawing to it
+        .with_bundle(WindowBundle::from_config_path(display_config_path))?
+        // A Processor system is added to handle loading spritesheets.
+        .with(
+            Processor::<SpriteSheet>::new(),
+            "sprite_sheet_processor",
+            &[],
+        )
+        // The renderer must be executed on the same thread consecutively, so we initialize it as thread_local
+        // which will always execute on the main thread.
+        .with_thread_local(RenderingSystem::<DefaultBackend, _>::new(
+            ExampleGraph::default(),
+        ));
+
+    let assets_dir = app_root.join("assets/");
+
+    let mut game = Application::new(assets_dir, Pong, game_data)?;
+    game.run();
 ```
 
 We've discovered Amethyst's root object: [Application][ap]. It binds the OS
 event loop, state machines, timers and other core components in a central place.
-Here we're creating a new `RenderBundle`, adding the `Pipeline` we created,
-along with our config, and building. There is also a helper function
-`with_basic_renderer` on `GameDataBuilder` that you can use to create your
-`Pipeline` and `RenderBundle`, that performs most of the actions above. This
-function is used in the full `pong` example in the `Amethyst` repository.
+Here we're creating a new `RenderingSystem`, adding the `ExampleGraph` we
+created, along with our config, and building.
```

Comment:
> and building

This sounds a little unclear to me. Could you try to rephrase that, please?

Reply:
Heh, I'll try to understand what the new code is actually doing and write something up :)